src/app/map-viewer/map-viewer.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Map } from 'leaflet';
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs/Subject';
+import { Observable } from 'rxjs/Rx';
+
+@Injectable()
+export class MapViewerService {
+  public map: Map;
+  public baseMaps: any;
+  public currentLat: number;
+  public currentLng:  number;
+  public altitude: number;
+  public accuracy: number;
+  hasLocation = false;
+  public center: any;
+
+  locationSource = new Subject<any>();
+
+  geoOptions = {
+      enableHighAccuracy: true,
+      maximumAge: 30000,
+      timeout: 27000
+  };
+  constructor() {
+    this.hasLocation = false;
+      if (navigator.geolocation) {
```

Comment:
To fit into your vision, which is: About Us page is the page to show all coders signed up on our web site.

Map section: all registered volunteers will show up on the map. There will be hundreds or thousands of dots.

Avatar section: do we still need this section? 
1. There will be hundreds or thousands of avatars
2. This is kind of duplicate of Volunteers page, which list all users (although with more detailed user information)